### PR TITLE
Support logging google analytics events with gtag.js

### PR DESCRIPTION
**Pull request by oyj9109**
Universal Analytics migrated analytics.js to gtag.js, so new GA customers will be using gtag function instead of ga function.
Refer to the links below for converting parameters from ga to gtag.
- UA migration guide: https://developers.google.com/analytics/devguides/migration/ua/analyticsjs-to-gtagjs
- News Tagging Guide: https://newsinitiative.withgoogle.com/training/datatools/ntg?platform=gtag&content=reader-revenue

### CODE REVIEW

File: src/runtime/google-analytics-event-listener.js [hunk 1]
```diff
@@ -57,8 +60,20 @@ export class GoogleAnalyticsEventListener {
       event.eventType,
       subscriptionFlow
     );
-    if (gaEvent) {
+    if (!gaEvent) {
+      return;
+    }
+
+    if (typeof this.win_.ga == 'function') {
```
[the remark this answers is not in the document]
Comment (oyj9109) [reply src/runtime/google-analytics-event-listener.js]: Agree. I found there's a util function for this, so changed it to use that.
https://github.com/subscriptions-project/swg-js/blob/462d1c9660c9d860a16c2ae58fc4b8fc466aa3ba/src/utils/types.js#L49

[the remark this answers is not in the document]
Comment (oyj9109) [reply src/runtime/google-analytics-event-listener.js]: Seems like javascript complier doesn't let me pass an undeclared property in a function.

**Comment by fikus** [comment]
It looks like this change supports logging an event with both `ga` and `gtag`. Would it make sense to instead prefer `gtag` if present but not call both functions? This would prevent double-logging in case an integrator somehow has both functions available.

**Review by adolgachev — APPROVED** [review]

**Comment by oyj9109** [comment]
> 
Google Analytics allows [Multiple tracking codes on web pages](https://support.google.com/analytics/answer/1032400?hl=en), so it's possible that two different versions of GA tags are configured on purpose. If both are present, anyway majority of the basic events GA supports such as page view, scroll, etc. will be double-logged, so I think it's safer to be consistent with them in terms of logging to both. Does that make sense?

**Comment by hartmannr76** [comment]
I think reading through those docs, it actually makes sense to just prefer gtag and fallback to ga as @fikus suggested. The multiple tracking codes seem to be more of having separate calls to `ga` to initialize the tracker, but they all end up using the same function

**Comment by oyj9109** [comment]
> I think reading through those docs, it actually makes sense to just prefer gtag and fallback to ga as @fikus suggested. The multiple tracking codes seem to be more of having separate calls to `ga` to [initialize the tracker](https://developers.google.com/analytics/devguides/collection/analyticsjs/creating-trackers#working_with_multiple_trackers), but they all end up using the same function

That could apply to between Universal Analytics (analytics.js) and Classic Analytics (ga.js) which use the same `ga` function. But it doesn't seem to work between `ga` and `gtag`. I tested with registering 2 different properties, one with analytics.js and the other with gtag.js, and calling `gtag` to see if it also supports the property registered by `ga`. It didn't, meaning that if the publisher set up multiple properties with both versions, they will lose metrics in the old version unless we explicitly call `ga` function.

**Comment by oyj9109** [comment]
> The difficult thing here is that it is very dependent on how the publisher is setting up their tagging. What may make the most sense is seeing what Analytics properties publishers are using today and figure out if it makes sense to fallback, send to both, or even remove ga entirely (since it doesn't really work without some configuration and UA properties will stop working next year) and establish that as the direction going forward.

Yeah I think we should make this change more carefully so agree with your suggestion. Let me do a quick research on the current status of GA set up for the existing publishers.

**Comment by oyj9109** [comment]
Just FTR, I did a quick research on the current `swg-basic.js` clients and most of them already migrated to `gtag.js`. They all properly migrated such that `ga` function can be still present but not sending events to google analytics, so either way should be fine for them.
Left a TODO for deprecating `ga` once universal analytics is down next year.

File: src/runtime/google-analytics-event-listener.js [hunk 2]
```diff
@@ -57,8 +60,21 @@ export class GoogleAnalyticsEventListener {
       event.eventType,
       subscriptionFlow
     );
-    if (gaEvent) {
+    if (!gaEvent) {
+      return;
+    }
+
+    // TODO(oyj9109): Remove it once universal analytics is deprecated in 2023.
```
[the remark this answers is not in the document]
Comment (oyj9109) [reply src/runtime/google-analytics-event-listener.js]: Done 👍

**Pull request merged**